src/main/java/com/ecore/roles/web/rest/RolesRestController.java

```diff
@@ -1,44 +1,52 @@
 package com.ecore.roles.web.rest;
 
 import com.ecore.roles.model.Role;
+import com.ecore.roles.service.MembershipsService;
 import com.ecore.roles.service.RolesService;
 import com.ecore.roles.web.RolesApi;
 import com.ecore.roles.web.dto.RoleDto;
 import lombok.RequiredArgsConstructor;
 import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
+import org.springframework.web.util.UriComponentsBuilder;
 
 import javax.validation.Valid;
+import java.net.URI;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.UUID;
 
 import static com.ecore.roles.web.dto.RoleDto.fromModel;
+import static java.util.stream.Collectors.toList;
 
 @RequiredArgsConstructor
 @RestController
 @RequestMapping(value = "/v1/roles")
 public class RolesRestController implements RolesApi {
 
     private final RolesService rolesService;
+    private final MembershipsService membershipsService;
 
     @Override
     @PostMapping(
             consumes = {"application/json"},
             produces = {"application/json"})
     public ResponseEntity<RoleDto> createRole(
-            @Valid @RequestBody RoleDto role) {
+            @Valid @RequestBody RoleDto role,
+            UriComponentsBuilder uriBuilder) {
+        RoleDto created = fromModel(rolesService.createRole(role.toModel()));
+        URI location = uriBuilder.path("/v1/roles/{roleId}").buildAndExpand(created.getId()).toUri();
         return ResponseEntity
-                .status(200)
-                .body(fromModel(rolesService.CreateRole(role.toModel())));
+                .created(location)
+                .body(created);
     }
 
     @Override
-    @PostMapping(
+    @GetMapping(
```

Comment:
In order to follow the Rest specification, of course, if the there isn’t any other security reason or URL length limitation